tests/helpers/test_update_coordinator.py

```diff
@@ -116,6 +116,39 @@ def update_callback():
     assert updates == [2]
 
 
+async def test_shutdown(
+    crd: update_coordinator.DataUpdateCoordinator[int],
+) -> None:
+    """Test async_shutdown for update coordinator."""
+    assert crd.data is None
+    await crd.async_refresh()
+    assert crd.data == 1
+    assert crd.last_update_success is True
+    # Make sure we didn't schedule a refresh because we have 0 listeners
+    assert crd._unsub_refresh is None
+
+    updates = []
+
+    def update_callback():
+        updates.append(crd.data)
+
+    _ = crd.async_add_listener(update_callback)
+    await crd.async_refresh()
+    assert updates == [2]
+    assert crd._unsub_refresh is not None
+
+    # Test shutdown through function
+    with patch.object(crd._debounced_refresh, "async_shutdown") as mock_shutdown:
+        await crd.async_shutdown()
+
+    # Test we shutdown the debouncer and cleared the subscriptions
```

Comment:
We should also call `async_fire_time_changed` to make sure it doesn't fire again to guard against future refactoring errors

Reply:
done